### PR TITLE
RemoveClassifiedEventActivity: check if calendar still exists

### PR DESCRIPTION
**Pull request by rullzer**
Thnx for spotting @go2sh

### CODE REVIEW

**Comment by ChristophWurst** [comment]
What about https://github.com/nextcloud/server/issues/13499#issuecomment-453411057?

**Review by ChristophWurst — DISMISSED** [review]
Diff makes sense :+1: 

Not tested yet, but trying to reproduce now

**Comment by go2sh** [comment]
I did update our instance to 15.0.1, but I was not able to reproduce it.

**Review by go2sh — APPROVED** [review]

**Comment by rullzer** [comment]
Mmmm this is weird. On my own setup the column names are correct.

**Comment by rullzer** [comment]
(i.e. this patch breaks them)

**Comment by rullzer** [comment]
maybe the problem comes from some events that do not have a proper principal uri set?

**Review by nickvergessen — CHANGES_REQUESTED** [review]
This does not work. The array does not have the table aliases on the entries

**Comment by ChristophWurst** [comment]
> This does not work. The array does not have the table aliases on the entries

Right 
bildschirmfoto von 2019-01-11 09-27-23

(upgrade to master)

File: apps/dav/lib/Migration/RemoveClassifiedEventActivity.php [hunk 1]
```diff
@@ -75,6 +75,10 @@ protected function removePrivateEventActivity(): int {
 		$result = $query->execute();
 
 		while ($row = $result->fetch()) {
+			if ($row['principaluri'] === null) {
+				continue;
```
Comment (nickvergessen) [inline apps/dav/lib/Migration/RemoveClassifiedEventActivity.php]: Actually we should delete those right away. But I will add a new repair step to check this.

**Review by nickvergessen — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]
Makes sense!

**Review by nickvergessen — APPROVED** [review]

File: apps/dav/lib/Migration/RemoveClassifiedEventActivity.php (hunk 1 above)
Comment (MorrisJobke) [reply apps/dav/lib/Migration/RemoveClassifiedEventActivity.php]: Why wasn't this added to the SQL statement? 😉

**Pull request merged**